api/src/utils/csv-utils/csv-header-configs.ts

```diff
@@ -369,3 +368,122 @@ export const getLookupIdCellValidator = (
     ];
   };
 };
+
+/**
+ * Get the taxon header cell validator.
+ *
+ * Rules:
+ *  1. The cell must be a valid ITIS TSN or scientific name
+ *  2. The cell must be a valid species from the provided taxons
+ *
+ * @template StaticHeaderType
+ * @param {TaxonMap} taxonMap The list of taxons
+ * @param {CSVCellValidatorOptions} [options] cell validator options
+ * @returns {*} {CSVCellValidator} The validate cell callback
+ */
+export const getTaxonCellValidator = (taxonMap: TaxonMap, options?: CSVCellValidatorOptions): CSVCellValidator => {
+  return (params) => {
+    const taxonIdentifierCell = params.cell;
+    const taxonHeader = params.header;
+
+    if (taxonIdentifierCell === undefined) {
+      if (options?.optional) {
+        return [];
+      }
+
+      return [
+        {
+          error: 'Cell is required',
+          solution: 'Use a valid ITIS TSN or scientific name',
+          header: taxonHeader,
+          cell: taxonIdentifierCell
+        }
+      ];
+    }
+
+    const taxon = taxonMap.get(String(taxonIdentifierCell));
+
+    // If a valid taxon
+    if (taxon) {
+      return [];
+    }
+
+    // If an invalid TSN
+    if (typeof taxonIdentifierCell === 'number') {
+      return [
+        {
+          error: 'Invalid ITIS TSN',
+          solution: 'Use a valid ITIS TSN',
+          header: taxonHeader,
+          cell: taxonIdentifierCell
+        }
+      ];
+    }
+
+    // If an invalid scientific name
+    if (typeof taxonIdentifierCell === 'string') {
+      return [
+        {
+          error: 'Invalid scientific name',
+          solution: 'Use a valid scientific name',
+          header: taxonHeader,
+          cell: taxonIdentifierCell
+        }
+      ];
+    }
+
+    return [
+      {
+        error: 'Invalid species',
+        solution: 'Expecting a valid ITIS TSN or scientific name',
+        header: taxonHeader,
+        cell: taxonIdentifierCell
+      }
+    ];
+  };
+};
+/**
+ * Get an array-cell validator, which applies a cell validator to each array-value parsed from the cells original value.
+ *
+ * Rules:
+ *  1. The cell must be a string, with optional values separated by a delimiter.
+ *  2. The individual cell values (after splitting by delimiter) must satisfy the cell validator.
+ *
+ * @param {csvCellValidator} csvCellValidator The cell validator to apply to each element in the array
+ * @param {CSVArrayCellValidatorOptions} options The array-cell validator options
+ * @returns {*} {CSVCellValidator} The validate cell callback
+ */
+export const getArrayCellValidator: CSVArrayCellValidator = (
```

Comment:
thought: Maybe this function should only accept two params: (csvCellValidator: CSVCellValidator, options: CSVArrayCellValidatorOptions). This would allow array validators to accept "array" cells of other types ie: numbers, booleans etc. That way the caller would have full control of the cell validation.

Reply:
And in the case of `getTaxonCellValidator` we have all the information needed to generate the CSVError`s. This would allow us to drop the `getNonEmptyStringCellValidator` from the internal logic of this function.